### ISSUE TITLE
Eliminate GCP key

### ISSUE THREAD

**Issue by chipzoller** [opencost/opencost#2311]
**Is your feature request related to a problem? Please describe.**
The environment variable `CLOUD_PROVIDER_API_KEY` is present by default in [several manifests](https://github.com/search?q=repo%3Aopencost%2Fopencost%20CLOUD_PROVIDER_API_KEY&type=code) with a hard-coded value. This is to increase user time-to-value and provide less work for them. However, this goes against basic security practices and as a result is being flagged and alerted on by various linters and scanners in CI. Defining sensitive information such as passwords, keys, and API tokens, even if deemed "safe" for public consumption, should be avoided.

**Describe the solution you'd like**
Eliminate the hard-coded `CLOUD_PROVIDER_API_KEY` value and template allowing users to bring their own. Provide documentation instructing users on how to create a key with the required permissions.

**Describe alternatives you've considered**
Use another service which doesn't require hard-coding API keys/tokens.

**Additional context**
Related to https://github.com/kubecost/cost-analyzer-helm-chart/issues/2767

**Comment by chipzoller** [comment]
It seems like this may be a larger topic. The value [here ](https://github.com/opencost/opencost/blob/b597e87dd5b79fcf8bec48f61894165f0dd4a994/pkg/env/costmodelenv.go#L400-L402)may not cope with the env var being undefined or cause disruptions in other places in the code. The `CLOUD_PROVIDER_API_KEY` env var should be optional and only required for users who want pricing from GCP. Users who are not using and will not use other cloud providers do not care about pricing from GCP and should not be need to supply superfluous data like this.

**Comment by AjayTripathy** [comment]
I'm supportive here, but we would want to make it clear in documentation and potentially even block startup on GCP without this key. I also want to call out that we provide the GCP key for evaluation purposes and it is tightly scoped to only look at billing data, so there isn't a security risk.

**Comment by chipzoller** [comment]
>I'm supportive here, but we would want to make it clear in documentation and potentially even block startup on GCP without this key.

Agreed, but blocking init probably doesn't make sense unless there's also a switch that enables GCP access (which isn't there today). If such a switch is present and I, as a user, don't throw it, I shouldn't also be required to provide an API key to a cloud I'm not going to use.

**Comment by martinhausio** [comment]
> I'm supportive here, but we would want to make it clear in documentation and potentially even block startup on GCP without this key. I also want to call out that we provide the GCP key for evaluation purposes and it is tightly scoped to only look at billing data, so there isn't a security risk.

There is always a security risk when creating service account keys. GCP Workload Identity supports granting access to the service account that is running the POD the appropriate permission scopes and the key itself is irrelevant if the service account has the appropriate billing API read permissions, provided the OpenCost code doesn't require one.

(See https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity for more information)

I would really appreciate the GCP key to be optional, as it should be irrelevant if the service account used with workload identity already has GCP Billing API permissions.

**Comment by AjayTripathy** [comment]
We do support workload identities. I guess I'm ok to make the key optional, but we will need a warning that pricing is inaccurate without the key

**Comment by akshayparseja** [comment]
are we removing the gcp api key from opencost??

**Comment by AjayTripathy** [comment]
@chipzoller am I wrong or did we finish out https://github.com/kubecost/cost-analyzer-helm-chart/issues/2767 ?

**Comment by chipzoller** [comment]
We did not.
